### PR TITLE
Add dconf functions for fix scripts

### PR DESCRIPTION
**Pull request by redhatrises**
#### Description:

- Adds dconf_lock function for configuring DConf locks and dconf_settings function for configuring
  dconf settings

#### Rationale:

- Need functions to simplify dconf configurations and make them less error prone

### CODE REVIEW

**Comment by redhatrises** [comment]
@matejak would you mind taking a look at this PR? Figured you would have some input into making these better.

File: shared/bash_remediation_functions/dconf_settings.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		echo "Aborting."
+		exit 1
+	fi
+
+	SETTINGSFILES=()
+
+	# Check for setting in any of the DConf db directories
+	SETTINGSFILES+=$(grep -r "\[${_path}]" "/etc/dconf/db/" | grep -v "distro\|ibus" | cut -d":" -f1)
+	# Replace possible slash '/' character so we could use it in sed expressions below
+	_path_esc=${_path//$'/'/$'\/'}
```
[the remark this answers is not in the document]
Comment (redhatrises) [reply shared/bash_remediation_functions/dconf_settings.sh]: @matejak nevermind

**Comment by redhatrises** [comment]
@matejak made suggested changes

File: shared/bash_remediation_functions/dconf_settings.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	then
+		[ ! -z ${DCONFFILE} ] || $(echo "" >> ${DCONFFILE})
+		echo "[${_path}]" >> ${DCONFFILE}
+		echo "${_key}=${_value}" >> ${DCONFFILE}
+	else
+		if grep -q "${_key}" ${SETTINGSFILES[@]}
+		then
+			sed -i "s/${_key}=.*/${_key}=${_value}/g" ${SETTINGSFILES[@]}
+		else
+			sed -i "\|\[${_path}]|a\\${_key}=${_value}" ${SETTINGSFILES[@]}
```
Comment (matejak) [inline shared/bash_remediation_functions/dconf_settings.sh]: I have tried that and found out that there may be existing config entries that look like `key = value` (i.e. with whitespaces), that would escape the substitution.
Would you be interested in adding test for this to the SSG test suite?

File: shared/bash_remediation_functions/dconf_lock.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Function to configure DConf locks for RHEL and Fedora systems.
+#
+# Example Call(s):
+#
+#     dconf_lock 'org/gnome/login-screen' 'banner-message-enable' 'local.d' 'banner'
+#
```
Comment (matejak) [inline shared/bash_remediation_functions/dconf_lock.sh]: It is not clear what this function is supposed to do, so I can't verify it's functionality.

File: shared/bash_remediation_functions/dconf_settings.sh (hunk 2 above)
Comment (redhatrises) [reply shared/bash_remediation_functions/dconf_settings.sh]: Haven't really played with code in the test suite, but sure.

File: shared/bash_remediation_functions/dconf_lock.sh (hunk 3 above)
Comment (redhatrises) [reply shared/bash_remediation_functions/dconf_lock.sh]: @matejak this function adds an entry to `/etc/dconf/db/${db}/locks/${file}` which prevents a user from changing the setting.

File: shared/bash_remediation_functions/dconf_lock.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	# Check sanity of the input
+	if [ $# -ne "4" ]
+	then
+		echo "Usage: dconf_lock 'dconf_path' 'dconf_setting' 'dconf_db' 'dconf_lockfile'"
+	  	echo "Aborting."
+		exit 1
+	fi
+
+	# Check for setting in any of the DConf db directories
+	LOCKFILES=($(grep -r "^/${_key}/${_setting}$" "/etc/dconf/db/" | grep -v "distro\|ibus" | cut -d":" -f1))
```
Comment (matejak) [inline shared/bash_remediation_functions/dconf_lock.sh]: `LOCKFILES` is just used to check whether the `grep` command has found anything, right? Therefore, making it an array is confusing.

File: shared/bash_remediation_functions/dconf_settings.sh (hunk 2 above)
Comment (matejak) [reply shared/bash_remediation_functions/dconf_settings.sh]: Actually, te test suite makes it possible to test remediations and testing the shared functionality alone is not supported ATM, so let's defer the testing. However, please take a look at those whitespaces.

File: shared/bash_remediation_functions/dconf_lock.sh (hunk 4 above)
Comment (redhatrises) [reply shared/bash_remediation_functions/dconf_lock.sh]: @matejak done

**Comment by matejak** [comment]
Great, I can't wait to see this functionality used in remediations! Thank you!

**Pull request merged**